packages/compass-e2e-tests/tests/time-to-first-query.test.ts

```diff
@@ -59,7 +60,12 @@ describe('Time to first query', function () {
 
   it('can open compass, connect to a database and run a query on a collection (second run onwards)', async function () {
     // start compass inside the test so that the time is measured together
-    compass = await beforeTests();
+
+    // TODO: this should work with firstRun left as undefined, but even though the
+    // preferences gets saved to disk, it gets the old value when opening
+    // compass again. The only way to get the saved value in a new compass seems
+    // to be if the whole process is relaunched.
+    compass = await beforeTests({ firstRun: false });
```

Comment:
I feel like I'm going crazy. The idea with these tests is that the first one makes a new user directory, so it is guaranteed to be the first run and therefore you see the welcome modal. That works as planned.

Then the second test (this one) would reuse the same user directory. ie, it should see the saved settings and therefore the app should know that the user already opted in. But even though the preferences got saved and the file on disk says that the user opted in, the welcome modal pops up. 

I've logged `console.log('shouldShowWelcomeModal', {showedNetworkOptIn, networkTraffic});` in shouldShowWelcomeModal and you can see that showedNetworkOptIn is false. But if you look at the file in the user dir it is true.

My last vague hunch left is that it is somehow cached in the test process? Seems crazy. But I get something similar when testing compass manually: remove the preferences file and it doesn't take effect until you kill the process and start over.

Most likely I'm just missing something dumb, though.

Reply:
I can work around it by always passing --showed-network-opt-in, but would love to understand what's going on.